include/ecs/component_pool.h

```diff
@@ -0,0 +1,87 @@
+#ifndef ECS_COMPONENT_POOL_H
+#define ECS_COMPONENT_POOL_H
+
+#include <vector>
+#include <unordered_map>
+
+#include "common.h"
+
+namespace arch::ecs {
+	class IComponentPool {
+	public:
+		virtual ~IComponentPool() = default;
+		virtual void on_entity_destroyed(EntityId entity_id) = 0;
+	};
+
+	template <typename T>
+	class ComponentPool : public IComponentPool {
+	public:
+		void insert(EntityId entity_id, T component);
+		void remove(EntityId entity_id);
+
+		T& get_data(EntityId entity_id);
+		void on_entity_destroyed(EntityId entity_id) override;
+
+	private:
+		std::vector<T> _component_array;
+		std::unordered_map<EntityId, size_t> _entity_to_index_map;
+		std::unordered_map<size_t, EntityId> _index_to_entity_map;
+
+		size_t poolSize{};
+	};
+
+	template <typename T>
+	void ComponentPool<T>::insert(EntityId entity_id, T component) {
+		assert_msg(

```

Comment:
Throw custom exceptions instead of just writing to spdlog::error and exiting

Reply:
K, I'll do that later